SwiftGit2/Repository.swift

```diff
@@ -9,7 +9,34 @@
 import Foundation
 import Result
 
-public typealias CheckoutProgressBlock = SG2CheckoutProgressBlock
+public typealias CheckoutProgressBlock = (String?, Int, Int) -> Void
+
+func SG2CheckoutProgressCallback(path: UnsafePointer<Int8>, completed_steps: Int, total_steps: Int, payload: UnsafeMutablePointer<Void>) -> Void {
+	if (payload != nil) {
+		let buffer = UnsafeMutablePointer<CheckoutProgressBlock>(payload)
+		let block = buffer.move()
+		payload.dealloc(1)
```

Comment:
Right, was thinking of a it as a completion for some reason. Also something to note is that this doesn't happen in the test.